### PR TITLE
Canonical animation for dropdown nav

### PR DESCRIPTION
**Pull request by OlgaLyubin**
Resolves #4579

**Overall change:** 
Added a 0.2s ease-out animation to canonical navigation on open/close.
To trigger animation in storybook, toggle the "Open" knob.

**Code changes:**
- Added CSS animation and updated stories.
- Separated dropdown into canonical and amp.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [x] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-navigation/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 6.0.0-alpha.18 | [PR#2847](https://github.com/bbc/psammead/pull/2847) Added animation to Dropdown Navigation |
```
Comment (DenisHdz) [inline packages/components/psammead-navigation/CHANGELOG.md]:
```suggestion
| 6.0.0-alpha.18 | [PR#2847](https://github.com/bbc/psammead/pull/2847) Add animation to Dropdown Navigation |
```

File: packages/components/psammead-navigation/src/DropdownNavigation/index.jsx [hunk 2]
```diff
@@ -26,7 +26,41 @@ const getStyles = dir => {
           padding-${direction}: ${GEL_SPACING};`;
 };
 
-export const Dropdown = styled.div`
+const StyledDropdown = styled.div`
+  background-color: ${C_EBON};
+  clear: both;
+
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
```
Comment (DenisHdz) [inline packages/components/psammead-navigation/src/DropdownNavigation/index.jsx]: Can the media query be placed at the end?

**Review by DenisHdz — COMMENTED** [review]
Since we are just toggling the visibility in the `StyledDropdown`, when closing the dropdown, you can still see the borders of the list items as they are visible.

dropdown-navigation

File: packages/components/psammead-navigation/src/DropdownNavigation/index.jsx [hunk 3]
```diff
@@ -26,7 +26,50 @@ const getStyles = dir => {
...
+    isOpen
+      ? `visibility: visible; height: ${height}px;`
+      : `visibility: hidden;`}
+
+  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
+    display: none;
+    visibility: hidden;
+  }
+
+  @media (prefers-reduced-motion: reduce) {
```
Comment (DenisHdz) [inline packages/components/psammead-navigation/src/DropdownNavigation/index.jsx:48]: Good one!

**Review by DenisHdz — APPROVED** [review]
This looks great now. Thank you for the changes :)

**Review by Bopchy — APPROVED** [review]
LGTM 👍

**Comment by PriyaKR** [comment]
Testing: Looks good except a small UX mismatch with the font size and line height for R to L language for eg: arabic at 240-359px the expected font size/line height is 20/26px but actual is 16/20px.

**Comment by PriyaKR** [comment]
According to the designs i could see the font size and line height change at 360px but actual it seems to be doing at 320px.

**Comment by tochwill** [comment]
> According to the designs i could see the font size and line height change at 360px but actual it seems to be doing at 320px.

Thanks @PriyaKR, taking a look!

This is based on the media queries that are defined in the gel-foundations package, that control all of the fonts that we import. I'll double check this today :)

**Comment by tochwill** [comment]
Hey @PriyaKR,

I've checked regarding fonts - we've taken these from the typography module in psammead, that specifies the groups according to the GEL specs: https://www.bbc.co.uk/gel/guidelines/typography#type-sizes

According to this we should be changing at 320px, and the font sizes and line heights will differ based on the script in use for that service.

**Comment by PriyaKR** [comment]
Confirmed with @laurapsn that the navigation should be using the font type size pica.

**Comment by tochwill** [comment]
> Confirmed with @laurapsn that the navigation should be using the font type size pica.

That's correct! I believe we are currently, or does something look off?

**Comment by PriyaKR** [comment]
For R to L languages the component doesn't have the actual text but the font sizes and line heights doesn't seem to be following the Pica type.

**Comment by tochwill** [comment]
> For R to L languages the component doesn't have the actual text but the font sizes and line heights doesn't seem to be following the Pica type.

Hmm, which service are you testing with? I just tried arabic and it seems to be correct. Could I take a look with you so I can see where the issue is?

**Comment by PriyaKR** [comment]
Looks good to me.

**Pull request merged**